<launch>

    <!-- launch settings, CHECK before launch -->
    <arg name="record_bag" default="true"/>
    <arg name="use_foxglove" default="true" />

    <!-- function settings -->
    <arg name="bag_folder" default="/home/user/localization-ws/src/localization-devel-ws/bag" />
    <!--                    RECORD                       -->
    <arg name='name_of_record_file' default='0210_omni' />

    <!-- functions -->
    <executable cmd="ros2 bag record --output $(var bag_folder)/$(var name_of_record_file) /odoo_googoogoo /raw_obstacles /imu/data_raw /final_pose /lidar_pose "  output="screen" if="$(var record_bag)"/>

    <node pkg="ekf" exec="ekf_node" name="ekf_node" output="screen">
        <param name="robot_parent_frame_id" value="map" />
        <param name="robot_frame_id" value="base_footprint" />
        <param name="/use_sim_time" value="true" />
    </node>

    <node pkg="lidar_localization_pkg" exec="lidar_localization" name="lidar_localization" output="screen">
        <param name="side" value="1"/>
        <param name="debug_mode" value="False"/>
        <param name="visualize_candidate" value="False"/>
        <param name="likelihood_threshold" value="0.002"/>
        <param name="consistency_threshold" value="0.9"/>
    </node>

    <include file="$(find-pkg-share foxglove_bridge)/launch/foxglove_bridge_launch.xml" if="$(var use_foxglove)">
        <arg name="port" value="8765" />
    </include>

    <include file="$(find-pkg-share lidar_localization_pkg)/launch/view_obstacles_launch.xml" />
    

</launch>